rust/xaynet-mobile/src/settings.rs

```diff
@@ -42,9 +45,12 @@ impl Settings {
         self.keys = Some(keys);
     }
 
-    /// Set the scalar use for masking
+    /// Set the scalar to use for masking
+    ///
+    /// # Panics
+    /// Panics if a `Scalar` cannot be constructed from the given `scalar`.
     pub fn set_scalar(&mut self, scalar: f64) {
-        self.scalar = scalar;
+        self.scalar = Scalar::from_primitive(scalar).unwrap()
```

Comment:
I would prefer it if we could avoid a potential `panic` here.  It's also not clear to me when it can fail. Maybe we can combine it with the `check` method.
for example 

```rust
#[derive(Clone, Debug)]
pub struct Settings {
    /// The Xaynet coordinator URL.
    url: Option<String>,
    /// The participant signing keys.
    keys: Option<SigningKeyPair>,
    /// The scalar used for masking.
    scalar: Result<Scalar>,
    /// The maximum possible size of a message.
    max_message_size: MaxMessageSize,
}

impl Settings {
    /// Create new empty settings.
    pub fn new() -> Self {
        Self {
            url: None,
            keys: None,
            scalar: Ok(Scalar::unit()),
            max_message_size: MaxMessageSize::default(),
        }
    }
    
    pub fn set_scalar(&mut self, scalar: f64) {
        self.scalar = Scalar::from_primitive(scalar)
    }
    
    pub fn check(&self) -> Result<(), SettingsError> {
        if self.url.is_none() {
            Err(SettingsError::MissingUrl)
        } else if self.keys.is_none() {
            Err(SettingsError::MissingKeys)
        else if self.scalar.is_err() {
            Err(SettingsError::Scalar...)
        } else {
            Ok(())
        }
    }
}
```

Reply:
it should only fail if the scalar is `< 0` or `NaN` or `Inf`. i agree that we should add this to the checks.

Reply:
thanks for pointing out @Robert-Steiner (I'd completely missed the `check` function when I was changing this). agree that we should follow a similar pattern to the other error variants here.

Reply:
Looks great 👍
we should check if we need to just the ffi layer that is used in `xayn`. we also need to generate a new c header file with cbindgen `v0.17.0`. it is described in the [readme](https://github.com/xaynetwork/xaynet/tree/master/rust/xaynet-mobile#generate-c-header-file) how to do this

Reply:
looks like the `xain_ai` FFI layer uses `v0.11.0` of `xaynet` rather than `master` so should be fine (thanks @Robert-Steiner for checking).
C header file regenerated - there is a change in `gcc` that affects the Linux command to compile FFI tests, updated the readme.